src/wsrep_provider_v26.cpp

```diff
@@ -423,14 +424,39 @@ namespace
         }
     }
     
-    wsrep_cb_status_t encrypt_cb(void*                 /* app ctx */,
-                                 wsrep_enc_ctx_t*      /*ctx*/,
-                                 const wsrep_buf_t*    /*input*/,
-                                 void*                 /*output*/,
-                                 wsrep_enc_direction_t /*direction*/,
-                                 bool                  /*final*/)
+    int encrypt_cb(void*                 app_ctx,
+                   wsrep_enc_ctx_t*      enc_ctx,
+                   const wsrep_buf_t*    input,
+                   void*                 output,
+                   wsrep_enc_direction_t direction,
+                   bool                  last)
     {
-        return WSREP_CB_SUCCESS;
+        assert(app_ctx);
+        wsrep::server_state& server_state(
+            *static_cast<wsrep::server_state*>(app_ctx));
+
+        wsrep::const_buffer key((unsigned char *)enc_ctx->key->ptr + 8, 
+                                enc_ctx->key->len - 8);
+        unsigned int key_version = *(unsigned int*)enc_ctx->key->ptr;
+        wsrep::const_buffer in(input->ptr, input->len);
+
+        try
+        {   
+            return server_state.server_service().do_crypt(&enc_ctx->ctx,
+                                                          key,
+                                                          key_version,
+                                                          enc_ctx->iv,
+                                                          in,
+                                                          output,
+                                                          direction == WSREP_ENC,
+                                                          last);
+        }
```

Comment:
There are several issues here:
1) provider should not know anything about key_version. server_service does. So the key must be passed to server service in its entirety and parsed there.
2) C-style casts
3) casting buffer to integer sometimes may bite back, no matter how certain we may be about alignment. Some platforms just freak out. memcpy() is the least one can do to ensure portability. (Better yet, if one deals with an integer, he may want to convert it to some canonical endianness before memcpy)
4) here's anonymous literal `8`. and the same literal is in a totally different file `server_state.cpp` and there is no apparent relation ship between them. This will cause troubles in maintenance.

I would suggest implementing two methods:
```
wsrep::const_buffer wsrep_key_serialize(const void* key, size_t size, unsigned int version)
```
and
```
wsrep_key_unserialize(wsrep::const_buffer& wk, void*& key, size_t& size, unsigned int& version);
```
side-by-side in a single compilation unit, most appropriately - `server_service` implementation - where they should be used.